model/riscv_step.sail

```diff
@@ -24,80 +136,111 @@ function step(step_no : int) -> bool = {
   minstret_write = None();
   minstreth_write = None();
 
-  let (retired, stepped) : (Retired, bool) =
-    match dispatchInterrupt(cur_privilege) {
-      Some(intr, priv) => {
-        if   get_config_print_instr()
-        then print_bits("Handling interrupt: ", interruptType_to_bits(intr));
-        handle_interrupt(intr, priv);
-        (RETIRE_FAIL, false)
-      },
-      None() => {
-        /* the extension hook interposes on the fetch result */
-        match ext_fetch_hook(fetch()) {
-          /* extension error */
-          F_Ext_Error(e)   => {
-            ext_handle_fetch_check_error(e);
-            (RETIRE_FAIL, false)
-          },
-          /* standard error */
-          F_Error(e, addr) => {
-            handle_mem_exception(Virtaddr(addr), e);
-            (RETIRE_FAIL, false)
-          },
-          /* non-error cases: */
-          F_RVC(h) => {
-            sail_instr_announce(h);
-            instbits = zero_extend(h);
-            let ast = ext_decode_compressed(h);
-            if   get_config_print_instr()
-            then {
-              print_instr("[" ^ dec_str(step_no) ^ "] [" ^ to_str(cur_privilege) ^ "]: " ^ BitStr(PC) ^ " (" ^ BitStr(h) ^ ") " ^ to_str(ast));
-            };
-            /* check for RVC once here instead of every RVC execute clause. */
-            if currentlyEnabled(Ext_Zca) then {
-              nextPC = PC + 2;
-              (execute(ast), true)
-            } else {
-              handle_illegal();
-              (RETIRE_FAIL, true)
-            }
-          },
-          F_Base(w) => {
-            sail_instr_announce(w);
-            instbits = zero_extend(w);
-            let ast = ext_decode(w);
-            if   get_config_print_instr()
-            then {
-              print_instr("[" ^ dec_str(step_no) ^ "] [" ^ to_str(cur_privilege) ^ "]: " ^ BitStr(PC) ^ " (" ^ BitStr(w) ^ ") " ^ to_str(ast));
-            };
-            nextPC = PC + 4;
-            (execute(ast), true)
-          }
-        }
-      }
+  let ((step_val, did_step), instbits) : ((Step, bool), option(xlenbits)) =
+    match step_state {
+      STEP_WAIT   => (step_wait_state(step_no, exit_wait), None()),
+      STEP_ACTIVE => step_active_state(step_no, exit_wait)
     };
 
-  tick_pc();
+  var stepped : bool = did_step;
```

Comment:
I believe you can have these named the same.

```suggestion
  var stepped : bool = stepped;
```

Reply:
The compiler does not like that:
`Cannot modify immutable let-bound constant or enumeration constructor stepped`